batch/batches/Delete/Engines/KDeletingEngine.php

```diff
@@ -88,6 +123,12 @@ public function run(KalturaFilter $filter)
 	{
 		KBatchBase::impersonate($this->partnerId);
 		$ret = $this->delete($filter);
+		self::$deletedRecordCount += $ret;
+		if((self::$deletedRecordCount % $this->deleteOperationBulkSize) == 0)
```

Comment:
Looking at the filters, we don't set pageSize = 500 explicitly
Also, I think if someone change a filter or if not all 500 return due to index, we will not sleep

Maybe we should cycle every time we passed 500
like:
if(((self::$deletedRecordCount / $this->deleteOperationBulkSize) % $this->deleteOperationBulkSize) >= 1)
{
	self::$deletedRecordCount = 0;
	sleep($this->deleteOperationBackOffInterval);
}

That way, no matter the pageSize or if not all objects returned we sleep every 500 objects

Reply:
I am not sure I follow this comment.
No one should set page size which is big it wont make sense as it might die on memory.
I dont really want to be aligned to page size to do the sleep necessarily, and if I do than I will be able to control it using the worker config.